### ISSUE TITLE
Installation on Ubuntu 16.04 and MySQL 5.7.15

### ISSUE THREAD

**Issue by perepechaev** [tony2001/pinba_engine#56]
Hi, i compiled mysql

```
apt source mysql-server-5.7
cd mysql-5.7-5.7.15
cmake 
```

and fetched pinba-engine-source

```
apt source pinba-engine-mysql-5.7
cd pinba-engine-mysql-1.1.0
./buildconf.sh
./configure --with-mysql=/tmp/mysql/mysql-5.7-5.7.15 --libdir=/usr/local/mysql/lib/plugin/
sudo make install
```

And after trying install plugin i get error:

```
mysql> INSTALL PLUGIN pinba SONAME 'libpinba_engine.so';
ERROR 1126 (HY000): Can't open shared library '/usr/local/mysql/lib/plugin/libpinba_engine.so' (errno: 2 /usr/local/mysql/lib/plugin/libpinba_engine.so: undefined symbol: mysql_malloc_service)
```

Also i am trying rebuild pinba with release-engine-1.2.0, release-engine-1.0.0 and get this error

I found bug in ubuntu tracker: https://bugs.launchpad.net/ubuntu/+source/pinba-engine-mysql/+bug/1590893 . But I don't know what I have to do

```
ldd -v /usr/local/mysql/lib/plugin/libpinba_engine.so
    linux-gate.so.1 =>  (0xb7775000)
    libevent-2.0.so.5 => /usr/lib/i386-linux-gnu/libevent-2.0.so.5 (0xb76b9000)
    libm.so.6 => /lib/i386-linux-gnu/libm.so.6 (0xb7664000)
    libJudy.so.1 => /usr/lib/i386-linux-gnu/libJudy.so.1 (0xb763a000)
    libc.so.6 => /lib/i386-linux-gnu/libc.so.6 (0xb7484000)
    libpthread.so.0 => /lib/i386-linux-gnu/libpthread.so.0 (0xb7467000)
    /lib/ld-linux.so.2 (0x800e3000)

    Version information:
    /usr/local/mysql/lib/plugin/libpinba_engine.so:
        libm.so.6 (GLIBC_2.0) => /lib/i386-linux-gnu/libm.so.6
        libc.so.6 (GLIBC_2.1.3) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.4) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.3.2) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.3.4) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.0) => /lib/i386-linux-gnu/libc.so.6
    /usr/lib/i386-linux-gnu/libevent-2.0.so.5:
        libpthread.so.0 (GLIBC_2.0) => /lib/i386-linux-gnu/libpthread.so.0
        libc.so.6 (GLIBC_2.3.4) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.3.2) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.1.3) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.15) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.1) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.4) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.7) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.17) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.0) => /lib/i386-linux-gnu/libc.so.6
    /lib/i386-linux-gnu/libm.so.6:
        ld-linux.so.2 (GLIBC_PRIVATE) => /lib/ld-linux.so.2
        libc.so.6 (GLIBC_2.1.3) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.0) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_PRIVATE) => /lib/i386-linux-gnu/libc.so.6
    /usr/lib/i386-linux-gnu/libJudy.so.1:
        libc.so.6 (GLIBC_2.1.3) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.4) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.0) => /lib/i386-linux-gnu/libc.so.6
    /lib/i386-linux-gnu/libc.so.6:
        ld-linux.so.2 (GLIBC_2.3) => /lib/ld-linux.so.2
        ld-linux.so.2 (GLIBC_PRIVATE) => /lib/ld-linux.so.2
        ld-linux.so.2 (GLIBC_2.1) => /lib/ld-linux.so.2
    /lib/i386-linux-gnu/libpthread.so.0:
        ld-linux.so.2 (GLIBC_2.1) => /lib/ld-linux.so.2
        ld-linux.so.2 (GLIBC_PRIVATE) => /lib/ld-linux.so.2
        libc.so.6 (GLIBC_2.3.2) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.1.3) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.1) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.0) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_2.2) => /lib/i386-linux-gnu/libc.so.6
        libc.so.6 (GLIBC_PRIVATE) => /lib/i386-linux-gnu/libc.so.6
```

```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.1 LTS
Release:    16.04
Codename:   xenial
$ uname -a
Linux site 4.4.0-45-generic #66-Ubuntu SMP Wed Oct 19 14:12:05 UTC 2016 i686 i686 i686 GNU/Linux
```

**Comment by perepechaev** [comment]
also i'm trying devel version with last commit 76d646e3dafb08d07080237ad200f085908929e9

**Comment by ValentinNikolaev** [comment]
confirm problem. the same problem
Ubuntu 16.04 and MySQL 5.7.16

**Comment by tony2001** [comment]
Same problem again and again.
The package was configured in one way and the sources you are using were configured in another way.
That's why I recommend building the server yourself - this way you'll be using the same sources and it'll work 100%.

**Comment by perepechaev** [comment]
Hy tony. I've rebuild mysql-server. And after that pinba-engine-plugin was rebuild too. 

`mysql> status
--------------
mysql  Ver 14.14 Distrib 5.7.16, for Linux (i686) using  EditLine wrapper

Connection id:		3
Current database:
Current user:		root@localhost
SSL:			Not in use
Current pager:		stdout
Using outfile:		''
Using delimiter:	;
Server version:		5.7.16 Source distribution`

But get other error: 
`mysql> INSTALL PLUGIN pinba SONAME 'libpinba_engine.so';
ERROR 1126 (HY000): Can't open shared library '/usr/local/mysql/lib/plugin/libpinba_engine.so' (errno: 2 /usr/local/mysql/lib/plugin/libpinba_engine.so: undefined symbol: mysql_malloc_service)`

**Comment by perepechaev** [comment]
Tony, what i am doing wrong?

**Comment by tchwpkgorg** [comment]
Hmm, why was this ticket closed? pinba-engine-mysql-5.7 is broken in Ubuntu 16.04.

**Comment by asakasinsky** [comment]
the same problem
